src/core/index.js

```diff
@@ -46,74 +51,50 @@ export default function SwaggerUI(opts) {
     buildTimestamp: BUILD_TIME,
   }
 
-  const domNode = opts.domNode
-  delete opts.domNode
+  const queryOptions = optionsFromQuery()(options)
+  let mergedOptions = mergeOptions({}, defaultOptions, options, queryOptions)
+  const storeOptions = storeOptionsFactorization(mergedOptions)
+  const InlinePlugin = inlinePluginOptionsFactorization(mergedOptions)
 
-  const { queryConfig, combinedConfig, storeConfig } = getConfigs(opts)
-
-  const inlinePlugin = ()=> {
-    return {
-      fn: combinedConfig.fn,
-      components: combinedConfig.components,
-      state: combinedConfig.state,
-    }
-  }
-
-  const store = new System(storeConfig)
-  store.register([combinedConfig.plugins, inlinePlugin])
 
+  const store = new System(storeOptions)
+  store.register([mergedOptions.plugins, InlinePlugin])
   const system = store.getSystem()
 
-  const downloadSpec = (fetchedConfig) => {
-    const localConfig = system.specSelectors.getLocalConfig ? system.specSelectors.getLocalConfig() : {}
-    const mergedConfig = deepExtend({}, localConfig, combinedConfig, fetchedConfig || {}, queryConfig)
-
-    // deep extend mangles domNode, we need to set it manually
-    if(domNode) {
-      mergedConfig.domNode = domNode
-    }
-
-    store.setConfigs(mergedConfig)
-    system.configsActions.loaded()
-
-    if (fetchedConfig !== null) {
-      if (!queryConfig.url && typeof mergedConfig.spec === "object" && Object.keys(mergedConfig.spec).length) {
-        system.specActions.updateUrl("")
-        system.specActions.updateLoadingStatus("success")
-        system.specActions.updateSpec(JSON.stringify(mergedConfig.spec))
-      } else if (system.specActions.download && mergedConfig.url && !mergedConfig.urls) {
-        system.specActions.updateUrl(mergedConfig.url)
-        system.specActions.download(mergedConfig.url)
+  const configURL = queryOptions.config ?? mergedOptions.configUrl
```

Comment:
Do we actually allow to use `config` instead of `configURL`?

I changed it to just get `configURL` from `mergedOptions` before, as that's the setting defined in documentation and the value from `queryOptions` should take precedence after merging with other options.

Reply:
Well, truthfully I don't know. According to the code, `queryOptions.config` is equivalent of standard config `configUrl` option. As this is purely a code consolidation I don't want to introduce any breaking changes.